core/src/test/java/com/linecorp/armeria/server/Http1ServerDelayedCloseConnectionTest.java

```diff
@@ -0,0 +1,158 @@
+        @Override
+        protected void configure(ServerBuilder sb) {
+            sb.idleTimeoutMillis(0);
+            sb.http(0);
+            sb.https(0);
+            sb.tlsSelfSigned();
+            sb.service("/close", (ctx, req) -> {
+                return HttpResponse.builder()
+                        .ok()
+                        .content("OK\n")
+                        .header(HttpHeaderNames.CONNECTION, "close")
+                        .build();
+            });
+        }
+    };
+
+    @Test
+    void shouldDelayDisconnectByServerSideIfClientDoesNotHandleConnectionClose() throws IOException {
+        final Random random = new Random();
+        final short localPort = (short) random.nextInt(Short.MAX_VALUE + 1);
+        try (Socket socket = new Socket("127.0.0.1", server.httpPort(),  null, localPort)) {
+            socket.setSoTimeout(100000);
+            final PrintWriter writer = new PrintWriter(socket.getOutputStream());
+            writer.print("GET /close" + " HTTP/1.1\r\n");
+            writer.print("\r\n");
+            writer.flush();
+
+            final BufferedReader in = new BufferedReader(new InputStreamReader(socket.getInputStream()));
+            assertThat(in.readLine()).isEqualTo("HTTP/1.1 200 OK");
+
+            String line;
+            boolean hasConnectionClose = false;
+            while ((line = in.readLine()) != null) {
+                if ("connection: close".equalsIgnoreCase(line)) {
+                    hasConnectionClose = true;
+                }
+                if (line.isEmpty() || line.contains(":")) {
+                    continue;
+                }
+                if (line.startsWith("OK")) {
+                    break;
+                }
+            }
+            final long readStartTimestamp = System.nanoTime();
+            final int readResult = in.read();
+            final long readDurationMillis = Duration.ofNanos(System.nanoTime() - readStartTimestamp).toMillis();
+
+            assertThat(hasConnectionClose).isTrue();
+            assertThat(readResult).isEqualTo(-1);
+
+            final long defaultHttp1ConnectionCloseDelayMillis = Flags.defaultHttp1ConnectionCloseDelayMillis();
+            assertThat(readDurationMillis).isBetween(
+                    defaultHttp1ConnectionCloseDelayMillis - 100,
+                    defaultHttp1ConnectionCloseDelayMillis + 1000
+            );
+
+            // Additional test to check that the socket is in the CLOSED state
+            if (SystemInfo.osType() == OsType.LINUX || SystemInfo.osType() == OsType.MAC) {
+                socket.close();
+                try(Socket reuseSock = new Socket()) {
+                    assertThatCode(() -> reuseSock.bind(new InetSocketAddress((InetAddress) null, localPort)))
+                            .doesNotThrowAnyException();
+                }
+            }
+        }
+    }
+
+    @Test
+    void shouldWaitForDisconnectByClientSideFirst() throws IOException {
```

Comment:
The `shouldWaitForDisconnectByClientSideFirst` test checks if the server sends `the Connection:close` header and waits for the client to close the connection first, rather than closing the connection immediately. It additionally checks if the server closes the connection after the client closes it.
On the other hand, the `shouldDelayDisconnectByServerSideIfClientDoesNotHandleConnectionClose` test verifies that if the client receives the `Connection: close header` but ignores it without attempting to close the connection, the close connection task scheduled by the server executes well waiting for the set time on the sever side before closing the connection.
Therefore, I wrote two tests because I think the purpose of the two tests and what they verify are different.

Reply:
I see, I think I missed that point in the last review cycle. Thanks for the explanation 👍